homeworks/03-data-structure/src/binary_tree.ts

```diff
@@ -0,0 +1,117 @@
+import { IBinaryTree, TreeNode } from "./interfaces";
+
+export enum TraverseType {
+  BFS,
+  INORDER_DFS,
+  POSTORDER_DFS,
+  PREORDER_DFS,
+}
+
+function throwError(): never {
```

Comment:
It doesn't work as an exhaustive check. Should be: `function assertNever(arg: never): never { /* */ }`.